lib/http3/common.c

```diff
@@ -1351,8 +1368,10 @@ void h2o_http3_send_h3_datagrams(h2o_http3_conn_t *conn, uint64_t flow_id, h2o_i
 {
     for (size_t i = 0; i < num_datagrams; ++i) {
         h2o_iovec_t *src = datagrams + i;
-        uint8_t buf[quicly_encodev_capacity(flow_id) + src->len], *p = buf;
+        uint8_t buf[quicly_encodev_capacity(flow_id) + 1 + src->len], *p = buf; // The '+ 1' is for Context-ID (see below)
+        flow_id = flow_id >> 2; // H3 quarter stream id (https://www.rfc-editor.org/rfc/rfc9297#section-2.1)
         p = quicly_encodev(p, flow_id);
+        p[0] = 0; p++; // H3 Context-ID for UDP frames is 0 (https://www.rfc-editor.org/rfc/rfc9298#section-5)
```

Comment:
`flow_id` is being quartered for every datagram, suggest moving outside the loop.

Suggest a block before the for loop
```c
if (conn->peer_settings.h3_datagram_rfc) {
    flow_id = flow_id >> 2; // H3 quarter stream id (https://www.rfc-editor.org/rfc/rfc9297#section-2.1);
}
```

```suggestion
        p = quicly_encodev(p, flow_id);
        if (conn->peer_settings.h3_datagram_rfc) {
            p[0] = 0;
            p++; // H3 Context-ID for UDP frames is 0 (https://www.rfc-editor.org/rfc/rfc9298#section-5)
        }
```

Reply:
This is better, thanks, will apply it